flair/datasets/sequence_labeling.py

```diff
@@ -1508,6 +1508,308 @@ def __init__(
+        manually downloaded the WSD datasets in UFSAC format to your machine.
+        Obtain the most recent datasets from https://drive.google.com/file/d/1Oigo3kzRosz2VjyA44vpJZ58tDFyLRMO and copy
+        any two of the datasets in a folder called 'wsd_ufsac'. Then rename the files in the folder to train.xml and
+        test.xml (optional) according to which file you want to train and test the model on. Then set the base_path
+        parameter in the constructor to the path to the parent directory where the 'wsd_ufsac' folder resides.
+        :param base_path: Path to the custom WSD corpus ('wsd_ufsac' folder) on your machine
+        :param in_memory: If True, keeps dataset in memory giving speedups in training.
+        :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
+        """
+        if type(base_path) == str:
+            base_path: Path = Path(base_path)
+
+        # column format
+        #
+        # since only the WordNet 3.0 version for senses is consistently available for all provided datasets we will
+        # only consider this version
+        #
+        # also we ignore the id annotation used in datasets that were originally created for evaluation tasks
+        #
+        # if the other annotations should be needed simply add the columns in correct order according
+        # to the chosen datasets here and respectively change the values of the blacklist array and
+        # the range value of the else case in the token for loop in the convert_ufsac_to_conll function
+
+        columns = {0: "text", 1: "lemma", 2: "pos", 3: "wn30_key"}
+
+        # this dataset name
+        dataset_name = self.__class__.__name__.lower()
+
+        # default dataset folder is the cache root
+        if not base_path:
+            base_path = Path(flair.cache_root) / "datasets"
+        data_folder = base_path / dataset_name
+
+        path_to_train_file = data_folder / "test.xml"
+
+        # check if data there
+        if not data_folder.exists():
+            log.warning("-" * 100)
+            log.warning(f'WARNING: UFSAC corpus not found at "{data_folder}".')
+            log.warning(
+                'Necessary data can be found here: "https://drive.google.com/file/d/1Oigo3kzRosz2VjyA44vpJZ58tDFyLRMO"'
+            )
+            log.warning("-" * 100)
+
+        # convert the files to CoNLL
+        convert_ufsac_to_conll(data_file=Path(data_folder / "train.xml"), encoding="latin-1")
+
+        if path_to_train_file.exists():
+
+            convert_ufsac_to_conll(data_file=Path(data_folder / "test.xml"), encoding="latin-1")
+
+            super(WSD_UFSAC, self).__init__(
+                data_folder,
+                columns,
+                tag_to_bioes=None,
+                encoding="latin-1",
+                in_memory=in_memory,
+                document_separator_token=None if not document_as_sequence else "-DOCSTART-",
+                train_file='train.xml',
```

Comment:
instead of hard-coding this, allow this to be specified through the constructor.